frontend/ui/uimanager.lua

```diff
@@ -478,6 +480,146 @@ end
 dbg:guard(UIManager, 'unschedule',
     function(self, action) assert(action ~= nil) end)
 
+--- @todo Stick this in base/ffi and reference here?
+local ffi = require("ffi")
+local C = ffi.C
+-- for ioctl header definition:
+local dummy = require("ffi/posix_h")
+local rtc = require("ffi/rtc_h") -- luacheck: ignore
+function UIManager:setWakeupAlarm(seconds_from_now, enabled)
+    enabled = (enabled ~= nil) and enabled or true
+    logger.dbg("setWakeupAlarm for", seconds_from_now, "seconds from now")
+    local t = ffi.new("time_t[1]")
+    t[0] = C.time(nil)
+    t[0] = t[0] + seconds_from_now
+
+    local ptm = ffi.new("struct tm") -- luacheck: ignore
+    ptm = C.gmtime(t)
+    self._wakeup_scheduled_ptm = ptm
+
+    local wake = ffi.new("struct rtc_wkalrm")
+    wake.time.tm_sec = ptm.tm_sec
+    wake.time.tm_min = ptm.tm_min
+    wake.time.tm_hour = ptm.tm_hour
+    wake.time.tm_mday = ptm.tm_mday
+    wake.time.tm_mon = ptm.tm_mon
+    wake.time.tm_year = ptm.tm_year
+    -- wday, yday, and isdst fields are unused by Linux
+    wake.time.tm_wday = -1
+    wake.time.tm_yday = -1
+    wake.time.tm_isdst = -1
+
+    wake.enabled = enabled and 1 or 0
+
+    local err
+    local rtc0 = C.open("/dev/rtc0", C.O_RDONLY)
```

Comment:
I'd possibly at least add `| O_NONBLOCK` in there, just because ;p.

i.e.,

```lua
local bor = bit.bor
local rtc0 = C.open("/dev/rtc0", bor(C.O_RDONLY, C.O_NONBLOCK))
```

Reply:
If you wrap it like this then it can be committed right from GH. ;-)
````
```suggestion
    local rtc0 = C.open("/dev/rtc0", C.O_RDONLY)
```
````

Reply:
Yeah, but ideally the bit.bor local should go way earlier ;p.

Good to know, though, I'd never bothered to look into the proper syntax for those ;).

Reply:
True, but I'll transfer the FFI stuff to something base/ffi/rtc.lua in any case. :-)